runtime/domain.c

```diff
@@ -688,7 +688,7 @@ static void domain_create(uintnat initial_minor_heap_wsize) {
 #endif
 
   caml_reset_young_limit(domain_state);
-
+  caml_domain_initialize_hook();
```

Comment:
This is called while holding the `all_domains` lock. It is better to wait that the lock is released. I would call `caml_domain_initialize_hook` right before the call to `caml_callback` inside `domain_thread_func`.

Reply:
That seems better indeed.